_rules/image-button-non-empty-accessible-name-59796f.md

```diff
@@ -56,6 +56,8 @@ Each target element has an [accessible name][] that is neither empty (`""`), nor
 
 There is a known combination of a popular browser and assistive technology that does not by default support `title` as an [accessible name][].
 
+An image button with non-empty `title` attribute but empty `alt` attribute results in a button with an empty accessible name on many browsers. There is a known popular browser that uses the `title` attribute as fallback even with empty `alt`, resulting in a button with `title` attribute value as accessible name. 
```

Comment:
an image and an image button are different things.  they have two different naming algorithms for a reason.  why are they being merged into once test?

Reply:
I agree with you; we were almost talking about the need of a clearer definition for "usable text string" in accessible name computation rules.

That said, this rule is specific for image button elements (we didn't merge image and image button elements int one single test).

As per [Inapplicable Example 4](https://www.w3.org/WAI/content-assets/wcag-act-rules/testcases/59796f/ebd0080bacb8debc7ad069072240657df38c3e2c.html)
> The img element is not a user interface component, and so is not tested for [Success Criterion 4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value).

that confirms that images are not part of this rule.

Reply:
In light of @scottaohara clarification at https://github.com/w3c/html-aam/issues/414#issuecomment-1178384158, I think we can rephrase this in the line of
> The naming algorithm (+link to relevant HTML AAM part) uses the first non-empty name, but some AU/AT stop at the first existing one, even if empty.

This would make it clear that it is not just some browsers quirks deciding what to do, but an actual bug in some implementation (and would explain why our examples are built following specs and may behave differently on some browsers).

---
Maybe we can also add a background note in the line of:
> Contrarily to images, empty `alt` does not make image button decorative. Image buttons are button and therefore exposed as interactive elements.

which is one on these things that is obvious once stated but maybe not so obvious before seeing it once…